core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/environment/network/SubnetSelector.java

```diff
@@ -50,11 +76,19 @@
                 } else {
                     subnetsToParse = source.getGatewayEndpointSubnetMetas();
                 }
-                Map<String, CloudSubnet> publicSubnetMetas = subnetsToParse.entrySet().stream()
-                    .filter(entry -> !entry.getValue().isPrivateSubnet())
-                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
 
-                endpointGatewayCloudSubnet = chooseSubnet(null, publicSubnetMetas, selectedAZ, false);
+                Map<String, CloudSubnet> publicSubnetMetas;
```

Comment:
**question:** What are the key `String` values in this `Map`? Are they the names of the cloud subnets?

**suggestion:** Consider naming this `publicSubnetsMetadata`?

Reply:
I believe they're the subnets ids. I didn't define the format of the Map, and am just using the format defined in `EnvironmentNetworkResponse.subnetMetas`.

Naming it `publicSubnetsMetadata` matches the pattern in `EnvironmentNetworkResponse`, which uses `Metas` instead of `Metadata`.